reflex/components/sonner/toast.py

```diff
@@ -27,6 +33,50 @@
 toast_ref = Var.create_safe("refs['__toast']")
 
 
+class ToastAction(Base):
+    """A toast action; rendered a button in the toast."""
+
+    label: str
+    on_click: Any
+
+
+@serializer
+def serialize_action(action: ToastAction) -> dict:
+    """Serialize a toast action.
+
+    Args:
+        action: The toast action to serialize.
+
+    Returns:
+        The serialized toast action with on_click formatted to queue the given event.
+    """
+    if action.on_click is not None:
+        events = []
+        on_click_specs = action.on_click
+        if not isinstance(on_click_specs, list):
+            on_click_specs = [on_click_specs]
+        for spec in on_click_specs:
+            if isinstance(spec, EventHandler):
+                specs = [call_event_handler(spec, lambda: [])]
+            elif isinstance(spec, type(lambda: None)):
+                specs = call_event_fn(spec, lambda: [])
+            else:
+                specs = [spec]
+            events.extend(format.format_event(s) for s in specs)
+        on_click = Var.create(
+            f"() => {{queueEvents([{','.join(events)}], socket); processEvent(socket)}}",
+            _var_is_string=False,
+            _var_is_local=False,
+        )
+    else:
+        on_click = Var.create("() => null", _var_is_string=False, _var_is_local=False)
```

Comment:
If we can pass this kind of var here, it would probably be possible to allow it for built-in event triggers too right ?

Reply:
Little known feature; we actually _do_ allow it if you set the `_var_type` to `EventChain`, it just gets rendered as is 

https://github.com/reflex-dev/reflex/blob/db47d39979ac62ad0571dc4aca9f00e90b9ee2d1/reflex/components/component.py#L497-L499

https://github.com/reflex-dev/reflex/blob/db47d39979ac62ad0571dc4aca9f00e90b9ee2d1/reflex/utils/format.py#L572-L581

Reply:
I'll have to remember that one 👍